### PR TITLE
Correct error handling and status checks

### PR DESCRIPTION
**Pull request by dustinblack**
## Changes introduced with this PR

Several changes to correct and improve error handling and process status checking.

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by jaredoconnell — APPROVED** [review]

File: arcaflow_plugin_pcp/pcp_plugin.py [hunk 1]
```diff
@@ -174,8 +181,8 @@ def start_pcp(
             pcp2csv_cmd.extend(metrics)
 
         max_retries = 1
-        pcp2csv_return = ("", "")
-        pcp2json_return = ("", "")
+        pcp2csv_status = ("", "")
+        pcp2json_status = ("", "")
```
Comment (dbutenhof) [inline arcaflow_plugin_pcp/pcp_plugin.py]: This looks odd. Below we have `pcp2csv_status, pcp2csv_return = run_oneshot_cmd(pcp2csv_cmd)`, which sets each value to a string from the tuple returned by the function. Yet here we're initializing each `*_status` to a tuple.

Apparently this doesn't really matter since your logic expects that the calls will be made, fixing the types and defining the `*_return` variants ... but this still looks odd.

File: arcaflow_plugin_pcp/pcp_plugin.py [hunk 2]
```diff
@@ -218,19 +225,16 @@ def start_pcp(
...
-            )
+        # Return the appropriate error condition after max_retries
+        if "error" in pcp2json_status:
+            return (pcp2json_status, pcp2json_return)
+        if "error" in pcp2csv_status:
+            return (pcp2csv_status, pcp2csv_return)
+
+        # Since the above for loop should always return either success or error,
+        # and should never come to its natural end, if we get here, something
+        # unexpected went wrong.
```
Comment (dbutenhof) [inline arcaflow_plugin_pcp/pcp_plugin.py:236]: A more pythonic way to trap "natural end" of a loop is with an `else` clause, if you want to push that assumption into code. Or if you really want the fallback of "anytime we get here" (e.g., if someone added a `break`), I'd rephrase this to something more like "exited the loop without returning".

Comment (webbnh) [reply arcaflow_plugin_pcp/pcp_plugin.py:236]: I don't really think using an `else` on the `for` loop would help anything, here.  But, I wouldn't mind a rephrase of this comment...eg., since the _loop_ will never result in an _error_ return (we _retry_ on errors...).

But, I would be more inclined to just delete this comment -- I think the code is pretty clear:  we return success from the loop or we retry; when we run out of retries, we return one of two errors; if none of that happens, we return an error pointing out that we had an "Unknown failure" (I might have said "unexpected", but...).

The only real shortcoming here is that, if by some crazy circumstance, we reach the catch-all error return, we don't give any context other than "it didn't work".  (We might be able to find the `pcp2csv_return` from the output at line 202, but we won't find the `pcp2json_return`, as near as I can tell.)

**Review by webbnh — APPROVED** [review]
Other than the items that Dave pointed out, it looks good to me.

File: arcaflow_plugin_pcp/pcp_plugin.py (hunk 1 above)
Comment (dustinblack) [reply arcaflow_plugin_pcp/pcp_plugin.py]: I don't like this either, and I'm sure there is a better approach, but the reason for the initialization is that there isn't a guarantee in the logic below that either `*_status` value will be set since they have some mutually-exclusive conditional states.

Comment (dustinblack) [reply arcaflow_plugin_pcp/pcp_plugin.py]: But one or the other or both _should_ be set by the logic.

Comment (webbnh) [reply arcaflow_plugin_pcp/pcp_plugin.py]: I _think_ that one of the implications of Dave's comment is that, if you want to retain them, these lines should be replaced with
```python
        pcp2csv_status, pcp2csv_return = ("", "")
        pcp2json_status, pcp2json_return = ("", "")
```
which is to say, they should match the calls to `run_oneshot_cmd()` which they are intended to stand in for.

Comment (dbutenhof) [reply arcaflow_plugin_pcp/pcp_plugin.py]: I don't think an initial definition for the `*_return` values is important for the logic, but, yeah, Webb's suggestion would work.

My point is that the `*_status` values should never be *tuples* ... they should be strings with `"success"` or `"error"`.
